Tests/Contour.Common.Tests/RabbitChannelSpecs.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Contour.Serialization;
+using Contour.Transport.RabbitMq;
+using Contour.Transport.RabbitMq.Internal;
+
+using Moq;
+
+using NUnit.Framework;
+
+using RabbitMQ.Client;
+using RabbitMQ.Client.Framing.v0_9_1;
+
+// ReSharper disable InconsistentNaming
+
+namespace Contour.Common.Tests
+{
+    [TestFixture]
+    [Category("Unit")]
+    public class RabbitChannelSpecs
+    {
+        [Test]
+        public void when_publish_message_should_choose_payload_converter()
+        {
+            Mock<IModel> modelMock = new Mock<IModel>();
+            modelMock.Setup(m => m.CreateBasicProperties())
+                .Returns(new BasicProperties());
+
+            Mock<IMessageLabelHandler> messageLabelHandlerMock = new Mock<IMessageLabelHandler>();
+
+            Mock<IBusContext> busContextMock = new Mock<IBusContext>();
+            busContextMock.SetupGet(ctx => ctx.MessageLabelHandler)
+                .Returns(messageLabelHandlerMock.Object);
+
+            var contentType = "application/protobuf";
+            var payloadConverterMock = this.CreateConverter(contentType);
+
+            Mock<IPayloadConverterResolver> payloadConverterResolverMock = new Mock<IPayloadConverterResolver>();
+            payloadConverterResolverMock.Setup(
+                pcr => pcr.ResolveConverter(It.IsAny<string>()))
+                .Returns(payloadConverterMock.Object);
+
+            var sut = new RabbitChannel(Guid.NewGuid(), modelMock.Object, busContextMock.Object, payloadConverterResolverMock.Object);

```

Comment:
Should the transport channel depend on the converter directly? Can the bus containing all content type configurations provide them to the channel?